src/pages/SubmissionHistory/SubmissionHistory.js

```diff
@@ -123,16 +135,35 @@ export const SubmissionHistory = () => {
                 paddingTop: '1em'
               }),
               columns: [
-                ...isActionMenuEnabled() ?
+                ...getConfig().isActionMenuEnabled ?
                   [{
                     size: { basis: 100, grow: 0 },
                     field: 'actions',
-                    headerRenderer: () => h(TextCell, ['Actions']),
-                    cellRenderer: () => {
-                      return div(
-                        { style: { textAlign: 'center' } },
-                        [icon('cardMenuIcon', { size: 24, onClick: () => { window.alert('TODO: go to actions menu') } })]
-                      )
+                    headerRenderer: () => h(TextCell, {}, ['Actions']),
+                    cellRenderer: ({ rowIndex }) => {
+                      return h(MenuTrigger, {
+                        'aria-label': 'Action selection menu',
+                        popupProps: {
+                          style: { left: '-20px' }
+                        },
+                        content: h(Fragment, [
+                          h(MenuButton, {
+                            style: { fontSize: 15 },
+                            disabled: isRunSetInTerminalState(paginatedPreviousRunSets[rowIndex].state),
+                            tooltip: isRunSetInTerminalState(paginatedPreviousRunSets[rowIndex].state) && 'Cannot abort a terminal submission',
+                            onClick: () => {
+                              cancelRunSet(paginatedPreviousRunSets[rowIndex].run_set_id)
+                              notify('success', 'Abort submission request submitted successfully',
```

Comment:
This notify should be moved inside `cancelRunSet` inside the try block otherwise I think this notify will always be displayed irrespective of whether abort was success or not 🤔

Reply:
Hm, I see what you mean. I had the success inside of `cancelRunSet` initially, but this caused a problem where when the page is rendered, there would be constant X amount of success messages depending on how many submissions a user had. This is weird considering the success is inside of an `onClick`, so I don't know why we get the notification on a refresh or initial page render.

Reply:
Ok, I think I resolved this. I'm not sure why I was seeing what I was seeing. I also added an additional check for the button being disabled when the state is also `Canceling`.